kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PaginationContext.java

```diff
@@ -17,21 +17,50 @@
 
 import static io.delta.kernel.internal.util.Preconditions.checkArgument;
 
+import io.delta.kernel.Meta;
 import java.util.Objects;
 import java.util.Optional;
 
 /** {@code PaginationContext} carries pagination-related information. */
 public class PaginationContext {
 
-  public static PaginationContext forPageWithPageToken(long pageSize, PageToken pageToken) {
+  public static PaginationContext forPageWithPageToken(
+      String tablePath, long tableVersion, long pageSize, PageToken pageToken) {
     Objects.requireNonNull(pageToken, "page token is null");
-    return new PaginationContext(pageSize, Optional.of(pageToken));
+    Objects.requireNonNull(tablePath, "table path is null");
+    checkArgument(
+        tablePath.equals(pageToken.getTablePath()),
+        "Invalid page token: table path does not match the requested table path. "
```

Comment:
`table path does not match the requested table path` -- can you make it clearer what does `table path` mean and what does `requested table path` mean?

You can do this by adding the word `token`: `Invalid page token: token table path does not match the requested table path`